django/db/migrations/state.py

```diff
@@ -403,6 +407,11 @@ def from_model(cls, model, exclude_rels=False):
                 elif name == "index_together":
                     it = model._meta.original_attrs["index_together"]
                     options[name] = set(normalize_together(it))
+                elif name == "indexes":
+                    indexes = model._meta.original_attrs["indexes"]
+                    for index in indexes:
+                        index.name  # Cache the property
```

Comment:
The design of this doesn't seem right. Contrary to the comment, the purpose seems to be that you need to set `index._name` which accessing `index.name` happens to do. That's not really intuitive. If this can't be remedied right now, the comment should be updated to reflect the true purpose.